src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcDsa.ImportExport.cs

```diff
@@ -0,0 +1,153 @@
+            byte[] d, int dLength);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcKeyCreateByExplicitParameters")]
+        internal static extern SafeEcKeyHandle EcKeyCreateByExplicitParameters(
+            ECCurveType curveType,
+            byte[] qx, int qxLength,
+            byte[] qy, int qyLength,
+            byte[] d, int dLength,
+            byte[] p, int pLength,
+            byte[] a, int aLength,
+            byte[] b, int bLength,
+            byte[] gx, int gxLength,
+            byte[] gy, int gyLength,
+            byte[] n, int nLength,
+            byte[] h, int hLength,
+            byte[] s, int sLength);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_GetECKeyParameters")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool GetECKeyParameters(
+            SafeEcKeyHandle key, 
+            bool includePrivate,
+            out IntPtr qx_bn, out int x_cb,
+            out IntPtr qy_bn, out int y_cb,
+            out IntPtr d_bn_not_owned, out int d_cb);
+
+        internal static void GetECKeyParameters(SafeEcKeyHandle key, bool includePrivate, ref byte[] Qx, ref byte[] Qy, ref byte[] d)
+        {
+            if (key == null || key.IsInvalid)
+            {
+                Debug.Assert(false);
+                throw new CryptographicException();
+            }
+
+            IntPtr Qx_bn, Qy_bn, d_bn_not_owned;
+            int Qx_cb, Qy_cb, d_cb;
+            if (!GetECKeyParameters(key, includePrivate, 
+                out Qx_bn, out Qx_cb, 
+                out Qy_bn, out Qy_cb, 
+                out d_bn_not_owned, out d_cb))
+            {
+                throw new CryptographicException();
+            }
+
+            using (SafeBignumHandle Qx_h = new SafeBignumHandle(Qx_bn, true))
+            using (SafeBignumHandle Qy_h = new SafeBignumHandle(Qy_bn, true))
+            using (SafeBignumHandle d_h = new SafeBignumHandle(d_bn_not_owned, false))
+            {
+                Qx = Crypto.ExtractBignum(Qx_h, Qx_cb);
+                Qy = Crypto.ExtractBignum(Qy_h, Qy_cb);
+                d = Crypto.ExtractBignum(d_h, d_cb);
+            }
+        }
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_GetECCurveParameters")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool GetECCurveParameters(SafeEcKeyHandle key, bool includePrivate,
+            out ECCurveType curveType,
+            out IntPtr Qx, out int x_cb,
+            out IntPtr Qy, out int y_cb,
+            out IntPtr d_bn_not_owned, out int d_cb,
+            out IntPtr P, out int P_cb,
+            out IntPtr A, out int A_cb,
+            out IntPtr B, out int B_cb,
+            out IntPtr Gx, out int Gx_cb,
+            out IntPtr Gy, out int Gy_cb,
+            out IntPtr n, out int n_cb,
+            out IntPtr h, out int h_cb,
+            out IntPtr S, out int S_cb);
```

Comment:
Yes probably. I was thinking of doing that in order to make some of the lifetime ownership more clear